[ansible-test] use newer container images

relrod opened the pull request:
##### SUMMARY
Change:
- Bump default, ansible-base, distro containers

Test Plan:
- CI, hopefully

Signed-off-by: Rick Elrod <rick@elrod.me>

<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->

- Test Pull Request

##### COMPONENT NAME

ansible-test

mattclay commented on test/lib/ansible_test/_data/completion/docker.txt:
```diff
@@ -1,13 +1,13 @@
 centos6 name=quay.io/ansible/centos6-test-container:1.17.0 python=2.6 seccomp=unconfined
 centos7 name=quay.io/ansible/centos7-test-container:1.17.0 python=2.7 seccomp=unconfined
-centos8 name=quay.io/ansible/centos8-test-container:1.17.0 python=3.6 seccomp=unconfined
+centos8 name=quay.io/ansible/centos8-test-container:1.20.0 python=3.6 seccomp=unconfined
```
We might as well make this and the other version 20 containers use 21 since that's the latest tag.

mattclay commented on test/integration/targets/incidental_setup_mysql_db/vars/Suse.yml:
```diff
@@ -4,3 +4,9 @@ mysql_packages:
+mysql_cleanup_packages:
+    - python3-base
+    - python3-mysqlclient
+    - libpython3_6m1_0
```
There may be collections which will need this same change to their tests.

sivel replied:
fwiw, I'm in the process of removing this incidental.

relrod committed:
[ansible-test] use newer container images

Change:
- Bump default, ansible-base, distro containers
- We do NOT add fedora33 yet, because it doesn't work right on Shippable
  due to an old kernel. This will be added post-AZP.

Test Plan:
- CI

Signed-off-by: Rick Elrod <rick@elrod.me>

mkrizek commented:
!waffling core_review

relrod commented:
@mattclay to re-review

I pulled out the f33 stuff for now into its own branch since we're blocked on AZP for it. I think we can merge this in the meantime though.

samdoran commented:
!waffling core_review

mattclay commented on test/lib/ansible_test/_data/completion/docker.txt:
```diff
@@ -1,13 +1,13 @@
-default name=quay.io/ansible/default-test-container:2.7.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=collection
-default name=quay.io/ansible/ansible-base-test-container:1.6.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=ansible-base
+default name=quay.io/ansible/default-test-container:2.9.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=collection
+default name=quay.io/ansible/ansible-base-test-container:1.7.0 python=3.6,2.6,2.7,3.5,3.7,3.8,3.9 seccomp=unconfined context=ansible-base
```
A changelog entry is needed for these updates.

Also, I'd backport them to stable-2.10. The other chagnes are probably OK to backport as well, but if not, they should be split into a separate PR.

relrod replied:
Updated changelog entry to call out those two containers. And I was planning to backport everything to 2.10.

mattclay replied:
The default containers also move from Python 3.9 RC1 to GA, so that should be included in the changelog.

relrod committed:
specifically call out default-test-container and ansible-base-test-container in changelog

Signed-off-by: Rick Elrod <rick@elrod.me>

samdoran commented:
!waffling needs_revision

relrod committed:
note python version bump

Signed-off-by: Rick Elrod <rick@elrod.me>
